wallet/psbt.go

```diff
@@ -0,0 +1,298 @@
+// Copyright (c) 2020 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package wallet
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
+	"github.com/btcsuite/btcutil/psbt"
+	"github.com/btcsuite/btcwallet/waddrmgr"
+	"github.com/btcsuite/btcwallet/wallet/txauthor"
+	"github.com/btcsuite/btcwallet/wallet/txrules"
+	"github.com/btcsuite/btcwallet/wtxmgr"
+)
+
+// FundPsbt creates a fully populated PSBT packet that contains enough inputs to
+// fund the outputs specified in the passed in packet with the specified fee
+// rate. If there is change left, a change output from the wallet is added.
+//
+// NOTE: If the packet doesn't contain any inputs, coin selection is performed
+// automatically. If the packet does contain any inputs, it is assumed that full
+// coin selection happened externally and no additional inputs are added. If the
+// specified inputs aren't enough to fund the outputs with the given fee rate,
+// an error is returned. No lock lease is acquired for any of the
```

Comment:
So the caller should be holding some sort of coin selection mutex before it calls this function? I guess things _are_ serialized through `CreateSimpleTx` partially, however I don't think that does any internal coin selection locking itself.

Reply:
I added another note to the godoc comment to mention that the global coin selection lock must be held.